lm_eval/__main__.py

```diff
@@ -103,6 +110,12 @@ def parse_eval_args() -> argparse.Namespace:
         default="INFO",
         help="Log error when tasks are not registered.",
     )
+    parser.add_argument(
```

Comment:
We should not store `args.huggingface_token` as a string, as this is very bad for privacy--a better option is to have a `--huggingface_login` flag which, if set, will log in using an environment variable `os.env['HUGGINGFACE_LOGIN_TOKEN']`

Alternatively, is there a reason we can't just expect people to log in separately if it's a required step, via HF's cli?

Reply:
I see, you're right. My thought was that this would be similar to how the hf cli works.

But maybe `--huggingface_login` is a good middle ground.

Reply:
I push a new commit of `huggingface_login`.
But after thinking about it, if we change to using environment variables due to privacy concerns, it might be better to directly let users log in with the command `huggingface-cli login --token $HUGGINGFACE_TOKEN`, as there is no significant difference between the two methods in essence. 
What do you think of it, should keep this field or remove it?

Reply:
This seems like a UI/UX question more than anything else. My current recommendation would be to just tell users to log in if they're trying to use private datasets and see if we get complaints.

Reply:
I agree, I think we shouldn't add this flag.

Reply:
Got it. Thanks, then can please close this PR first~ I will submit new changes later

Reply:
as an unhelpful outsider... cli login is not ideal and working around this case now wishing there were a way to pass in the credential (which we can mitigate, just saying if youre concerned with greater UX coverage... here is one perspective)

Reply:
@backnotprop why is expecting people to log in via the cli before using the library not ideal for you?

Reply:
Merging this PR without the `--huggingface_login` flag. Would be curious to hear the usecase that prevents the cli login prior to usage of the library